shaders/mesh_shader_culling/mesh_shader_culling.mesh

```diff
@@ -0,0 +1,115 @@
+ * limitations under the License.
+ */
+
+#extension GL_EXT_mesh_shader: require
+
+layout(local_size_x = 64, local_size_y = 1, local_size_z = 1) in;
+layout(triangles, max_vertices = 64 * 4, max_primitives = 124 * 2) out;
+
+// Example of how to read shared data from its associated task shader:
+// 1) the same structure must be defined (name can vary however):
+
+struct SharedData
+{
+    vec4  positionTransformation;
+    int   N;
+    int   meshletsNumber;
+    float subDimension;
+    float cullRadius;
+};
+
+// 2) using the following command to "establish the connection" (variable name can vary):
+
+taskPayloadSharedEXT SharedData sharedData;
+
+// Example of a default output layout and location format:
+
+layout (location=0) out vec4 perVertex[];
+layout (location=1) perprimitiveEXT out vec4 perPrimitiveEXT[];
+layout (location=3) out vec3 outColor[];
+
+// Define some constants in order to create an identical meshlet square, using two triangles:
+// 1) a simplified vertex data, using vec2 format
+
+const vec2 unitVertex_0 = vec2(-0.5f,  0.5f);
+const vec2 unitVertex_1 = vec2( 0.5f,  0.5f);
+const vec2 unitVertex_2 = vec2( 0.5f, -0.5f);
+const vec2 unitVertex_3 = vec2(-0.5f, -0.5f);
+
+// 2) index data associated with the vertex data, triangle defined using right-hand rule
+
+const uvec3 unitPrimitive_0 = uvec3(0, 1, 2);
+const uvec3 unitPrimitive_1 = uvec3(2, 0, 3);
+
+void main()
+{
+    // 1) The first variable means: the total number of vertices to be created:
+    // 2) The second variable means total number of primitives to be created (at least 3 index per primitive):
+
+    SetMeshOutputsEXT( sharedData.meshletsNumber * 4, sharedData.meshletsNumber * 2);
+
+    // Creating meshlets:
+    for (uint k = 0; k < sharedData.meshletsNumber; k++)
+    {
+        // processing index, "breaking" a linear index into a coordinate index
+        // (i.e., (i, j) means the ith column and jth row)
+
+        float j = floor(k / sharedData.N);   // calculating which row this meshlet belongs to
+        float i = k - j * sharedData.N;      // calculating which column this meshlet belongs to
+
+        // translate the identical meshlet to its targeted position:
+
+        vec2 deltaPosition_xy = ( 2.0f * vec2( i, j ) - sharedData.N - 1.0f ) * sharedData.subDimension;
+
+        // this is the displacement with respect to the origin, displacement is used in cull logic:
+
+        vec4 displacement = vec4(deltaPosition_xy, 0.0f, 1.0f);
+
+        // the actual position of each meshlet:
+
+        vec4 position = displacement + sharedData.positionTransformation;
+        float squareRadius = position.x * position.x + position.y * position.y;
+
+        // Cull Logic: only is the meshlet center position is within the view circle defined by the cull radius,
+        // then this very meshlet will be generated. Otherwise, meshlet will NOT be generated.
+
+        if (squareRadius < sharedData.cullRadius * sharedData.cullRadius)
```

Comment:
If I'm reading this right then some elements of `gl_MeshVerticesEXT[]` and `gl_PrimitiveTriangleIndicesEXT[]` are not written to depending on this culling logic, is that right? However, `SetMeshOutputsEXT()` is always the same value regardless of the culling logic?

Is the implication then that the consumer of `gl_MeshVerticesEXT` and `gl_PrimitiveTriangleIndicesEXT` are then reading either (a) some  undefined values (because there were no writes to some of the elements), or (b) defined-to-be-something (zero?) values? 

Basically, I'm wondering if this is well-defined -- and also if this could really be called _culling_ when it seems more like the same number of primitives are always output, it's just than when 'culled' they are set up as degenerate primitives (thus _processed_ but not actually visible).